### PR TITLE
Epmcuxda 6668 password change

### PR DESCRIPTION
**Pull request by SerafimovichEugene**
### What does this PR do?

changePassword method allows to change user's password

### How should I test this?

`List of steps to reproduce, corroborate or tests to run. Write this section clear enough so that external users can also follow it and test the fix or feature.`

### Jira ticket related to this PR

- [EPMCUXDA-6668](https://jira.epam.com/jira/browse/EPMCUXDA-6668)

### Link to branch and/or PR's related

`Just remove section if it isn't related to anything`

- https://github.com/abahachuk/ptushki-backend/pull/XXX
- https://github.com/abahachuk/ptushki-frontend/pull/XXX
- https://github.com/abahachuk/ptushki-mobile/pull/XXX

#### Type of change

`What types of changes does your code introduce? Put an 'x' in all the boxes that apply:`

- [x] Adds new features.
- [ ] Changes existing functionality.
- [ ] Deprecates features.
- [ ] Removes features.
- [ ] Fixes any bug.
- [ ] Other: security, build process, infrastructure, tests, docs, etc.

### Screenshots (if appropriate)

`Screenshots of the feature or bug if available.`

### Questions (if any)

`If exists, to another developer or reviewer.`

### CODE REVIEW

File: src/controllers/auth-controller.ts [hunk 1]
```diff
@@ -22,6 +23,7 @@ export default class AuthController extends AbstractController {
     this.router.post('/logout', this.logout);
     this.router.post('/login', this.login);
     this.router.post('/refresh', this.refresh);
+    this.router.post('/changePassword', authRequired, this.changePassword);
```
Comment (mishelen) [inline src/controllers/auth-controller.ts]: camelCase variant not so good variant as url

File: src/controllers/auth-controller.ts [hunk 2]
```diff
@@ -99,6 +101,25 @@ export default class AuthController extends AbstractController {
     }
   };
 
+  private changePassword = async (req: Request, res: Response) => {
+    const { oldPassword, newPassword } = req.body;
+    const user = req.user as User;
+    if (!oldPassword || !newPassword || !user) {
+      return res.status(400).end();
+    }
+    const isPasswordCorrect = user ? await isCorrect(oldPassword, user.salt, user.hash) : false;
```
Comment (mishelen) [inline src/controllers/auth-controller.ts]: redundant ternary expression, `user` exists here

**Comment by mishelen** [comment]
@SerafimovichEugene next time when you fill PR just remove unused sections and comments, dont be lazy, thanks!

File: src/controllers/auth-controller.ts (hunk 2 above)
Comment (dzmitrypanamarenka) [reply src/controllers/auth-controller.ts]: yep, noted it too, you checked user before

Comment (dzmitrypanamarenka) [reply src/controllers/auth-controller.ts]: yep, noted it too, you checked user before

Comment (SerafimovichEugene) [reply src/controllers/auth-controller.ts]: fixed

File: src/controllers/auth-controller.ts (hunk 1 above)
Comment (SerafimovichEugene) [reply src/controllers/auth-controller.ts]: '/change-password' was applied

**Pull request merged**